config/api.go

```diff
@@ -56,8 +56,11 @@ type SeedConsumer struct {
 	SeedProducer
 	// CallbackURL represents the URl to call back
 	CallbackURL *url.URL
-	// Channels represents which channel this consumer listens to
+	// Channel represents which channel this consumer listens to
 	Channel string
+	// IsPullBased represents whether the consumer is pull based or not,
+	// default value is false, hence push based
+	IsPullBased bool
```

Comment:
Typically property names does not start with a verb such as `is` or `has`; that is more for behaviors/methods/functions. Just PullBased would be sufficient IMHO

Reply:
Bhaia, as we discussed, I tried to use `ConsumerType` (the way `JobStatus` currently works) instead of using a boolean flag. I faced one challenge about where to keep the [`ConsumerType`](https://github.com/newscred/webhook-broker/pull/140/files#diff-2f690a05792c7f0a5738fb4e4bdc87dc2bdb8f188a227b88fccf271e09adb44cR16) type. 
The main challenge orginates from the fact that unlike `JobStatus`, both `config` package and `Storage/data` package  will use `ConsumerType`. 
Several possible options came to my mind: 
1. Keeping it in  `storage/data/consumer.go` which results in circular import. 
2. Creating a separate package for `ConsumerType` which seemed to be a bit overkill to me.
3. Keeping it in `config` package.

I chose option 3. Please suggest me if there is a better place to keep it.